File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j = 0;
        for(int i = 1; i < nums.size(); i++){
            if(nums[i] != nums[j]){
                j++;
                nums[j] = nums[i];
            }
            
        }
        return j +1;


        // set<int> st;
        // for(int i = 0 ; i< nums.size(); i++){
        //     st.insert(nums[i]);
        // }
        // // vector<int>ans;
   
        
        // for(int i = 0; i< nums.size(); i++){
        //     nums.push_back(0);
        // }

        // int cnt = 0;
        // int size = st.size()-1;

        // for(auto it : st){
        //     cout<<it<<" ";
        //     if(cnt < size)
        //     nums[cnt++] = it;
        //     cout<< nums[cnt];

        // }
        // return size;
        
    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        // without space
        int size = nums.length-1;
        if(size == 0) return  1;
        int i = 0; 
        int j = 1;
        
        int cnt = 1;
        while(j <= size){
           if(nums[j] != nums[i]){
            cnt++;
            nums[i+1] = nums[j];
            i = i+1;
           } else{
            j++;
           }
        }
        return cnt;


        // with Extra Space

        // HashSet<Integer> st = new LinkedHashSet<>();
        // for(int i = 0; i< nums.length; i++){
        //     st.add(nums[i]);
        // }
        // int i = 0;
        // for(int val : st){
        //     nums[i++] = val;
        // }
        // return st.size();
    }
}